README.md

```diff
@@ -1,11 +1,158 @@
+
+- `name` is the human readable name of the period, according to what we dispaly in DHIS2 Data Visualizer, i.e. `Week 1 - 2014-13-03 - 2015-01-04` for a weekly period or `2015` for a yearly period. 
+
+## getNowInCalendar
+
+- `getNowInCalendar` returns a ZonedDateTime in the calendar for today's date. 
+
+> ToDo: we should also return the date stringified into `yyyy-MM-dd` since this is the most common usecase and it would help clients not to have to do the conversion manually.
+
+## Upcoming methods
+
+There are new methods that will be added soon (in the alpha release right now), such as `getFixedPeriodByDate` (which, given a date in a calendar, it returns what fixed period the date falls in)  and `getAdjacentFixedPeriods` (that facilitates being able to navigate to next and previous periods).
+
+> These new methods will be documented when merged into main branch, but this gives an idea about the general approach we want to follow with th library; if you encounter a use case that you think it should be included in the library then please get in touch to discuss whether it should live there, or at the consumer level.
+
+# Special cases and considerations with Periods logic
+
+There are some special DHIS-2 specific cases when doing period calcualtions that we handle. The specs for these are documented in the tests and cucumber feature files. Some of these are:
+
+- Dealing with the 13th month in the Ethiopian calendar: the Ethiopic calendar has 13 months: 12 months of 30 days, then the 13th month has 5 or 6 days depending on whether it's a leap year or not. When generating periods for bi-weekly or less (bi-weekly, weekly, daily) then the 13th month is displayed. But when we are generating periods for Monthly or larger, we do not display the 13th month. The backend then makes the decision about where to include any data for the 13th month, so when doing monthly analytics, the 13th month will be lumped with the following month's data.
+
+- The period IDs are dictated by the Backend, and they use the gregorian month names, for example, `QuarterlyNov` or `QuarterlyApr`. In the frontend, in a non-Gregorian calendar, `Nov` will be "translated" into the 11th month of the current calendar system (Hamle in Ethiopic). Ideally, we will have Period IDs that are calendar-agnostic in the future, but this works for now as a solution.
+
+- From Frontend perspective, we always send a date in the calendar system. It is formatted in an iso-like format (`yyyy-MM-dd`). It is "iso-like", because the date is actually in the calendar system rather than the gregorian iso8601 calendar. So when a system set to Ethiopic, sends `2015-01-01` to the Backend, it means the year 2015 in Ethiopic (which is 2022 in Gregorian). This is what the Backend expects, and if there is a need for a conversion, then it happens in the backend.
```

Comment:
```suggestion
- From the frontend perspective, we always send a date in the calendar system. It is formatted in an ISO-like format (`yyyy-MM-dd`). It is "ISO-like", because the date is actually in the calendar system rather than the Gregorian ISO8601 calendar. So when a system set to Ethiopic, sends `2015-01-01` to the backend, it means the year 2015 in Ethiopic (which is 2022 in Gregorian). This is what the backend expects, and if there is a need for a conversion, then it happens in the backend.
```